api.go

```diff
@@ -275,17 +266,28 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source CRUD) *r
 		}
 	})
 
-	api.router.Handle("GET", baseURL+"/:id", func(w http.ResponseWriter, r *http.Request, params map[string]string) {
-		info := requestInfo(r, api)
-		c := api.contextPool.Get().(APIContexter)
-		c.Reset()
-		api.middlewareChain(c, w, r)
-		err := res.handleRead(c, w, r, params, *info)
-		api.contextPool.Put(c)
-		if err != nil {
-			api.handleError(err, w, r)
-		}
-	})
+	if _, ok := source.(CRUDFindOne); ok {
+		api.router.Handle("OPTIONS", baseURL+"/:id", func(w http.ResponseWriter, r *http.Request, _ map[string]string) {
+			c := api.contextPool.Get().(APIContexter)
+			c.Reset()
+			api.middlewareChain(c, w, r)
+			w.Header().Set("Allow", "GET,PATCH,DELETE,OPTIONS")
```

Comment:
Actually there are two strings: one for the collection and another one for the item...
Here is the full logic, written very procedurally:
```go
allowedForCollection := [4]string{}
	allowedForItem := [4]string{}

	allowedForCollection[0] = http.MethodOptions
	allowedForItem[0] = http.MethodOptions

	if _, ok := source.(CRUDFindOne); ok {
		allowedForCollection[1] = http.MethodGet
		allowedForItem[1] = http.MethodGet
	}

	if _, ok := source.(CRUDCreate); ok {
		allowedForCollection[2] = http.MethodPost
	}

	if _, ok := source.(CRUDUpdate); ok {
		allowedForCollection[3] = http.MethodPatch
		allowedForItem[2] = http.MethodPatch
	}

	if _, ok := source.(CRUDDelete); ok {
		allowedForItem[3] = http.MethodDelete
	}
```